pkg/sdk/testint/errors_integration_test.go

```diff
@@ -0,0 +1,146 @@
+package testint
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/stretchr/testify/assert"
+)
+
+func schemaObjectShowByIDWrapper[T any](showByIdFn func(context.Context, sdk.SchemaObjectIdentifier) (*T, error)) func(context.Context, sdk.SchemaObjectIdentifier) error {
+	return func(ctx context.Context, id sdk.SchemaObjectIdentifier) error {
+		_, err := showByIdFn(ctx, id)
+		return err
+	}
+}
+
+func schemaObjectWithArgumentsShowByIDWrapper[T any](showByIdFn func(context.Context, sdk.SchemaObjectIdentifierWithArguments) (*T, error)) func(context.Context, sdk.SchemaObjectIdentifierWithArguments) error {
+	return func(ctx context.Context, id sdk.SchemaObjectIdentifierWithArguments) error {
+		_, err := showByIdFn(ctx, id)
+		return err
+	}
+}
+
+func TestInt_ShowSchemaObjectInNonExistingDatabase(t *testing.T) {
+	doesNotExistOrNotAuthorized := sdk.ErrObjectNotExistOrAuthorized.Error() // Database '\"non-existing-database\"' does not exist or not authorized
+	doesNotExistOrOperationCannotBePerformed := "Object does not exist, or operation cannot be performed"
+
+	testCases := []struct {
+		ObjectType  sdk.ObjectType
+		ExpectedErr string
+		ShowFn      func(context.Context, sdk.SchemaObjectIdentifier) error
+	}{
+		// Only object types that use IN SCHEMA in their ShowByID implementation
+		{ObjectType: sdk.ObjectTypeTable, ExpectedErr: doesNotExistOrOperationCannotBePerformed, ShowFn: schemaObjectShowByIDWrapper(testClient(t).Tables.ShowByID)},
```

Comment:
omg. are the errors for the given objects at least consistent for unexisting db/schema? (it's hard for me to compare on GH)

Reply:
The second thought: maybe after such a test, we could make the error returned from the show query consistent throughout the SDK? (so that on the resource level we can always check for the same error). And of course I don't mean we should do it now, but this is a nice improvement to document.

Reply:
I think they are consistent (as far as I remember, I copy-pasted the whole test for the database and didn't have to change much). Yeah, we can make them as one error. The one thing that I didn't like during this whole testing was that we already do this for the error `does not exist or not authorized`; I think a better solution would be to enclose the error as it is right now but somehow unwrap it when returning errors in resources. That way, the user will get more meaningful errors instead of generic ones. We could also benefit from this when getting logs from users. That's just my thought because, as far as I remember, I had many many errors during the testing, and unwrapping could maybe speed debugging up a bit (it's still not that bad because the original errors are visible in the SQL logs).

Reply:
let's add this to the Sdk improvements and we'll handle it after v1

Reply:
Both seem valid, I would be going by the SDK improvements probably this week, so maybe I will split the whole epic into smaller ones, will see. It's fine to keep it in the current one for now.